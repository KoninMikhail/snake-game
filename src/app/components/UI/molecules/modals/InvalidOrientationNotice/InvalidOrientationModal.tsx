import {
    StyledInvalidOrientationModalContainer,
    StyledInvalidOrientationModalContent,
    StyledInvalidOrientationNoticel,
} from './style/StyledInvalidOrientationNoticel';
import { GlassesIcon } from '@ui/atoms/assets/icons/GlassesIcon/GlassesIcon';
import { PrimaryFill } from '@ui/atoms/assets/fills/PrimaryFill/PrimaryFill';
import { FadeAnimation } from '@ui/atoms/assets/animations/FadeAnimation/FadeAnimation';
import { PrimaryTextShadow } from '@ui/atoms/assets/shadows/text/PrimaryTextShadow/PrimaryTextShadow';
import { Caption } from '@ui/atoms/typography/caption/Caption/Caption';
import { IInvalidOrientationModal } from '@ui/molecules/modals/InvalidOrientationNotice/InvalidOrientationModal.interface';

/**
 * @name InvalidOrientationModal
 * @author Mikhail Konin <dev.konin@gmail.com>
 *
 * Use to ready a message about incorrect screen orientation.
 *
 * @param children - message for user
 * @returns {JSX.Element}
 */
export const InvalidOrientationModal = ({ message }: IInvalidOrientationModal) => (
    <StyledInvalidOrientationNoticel>
        <FadeAnimation direction={'in'}>
            <StyledInvalidOrientationModalContainer>
                <StyledInvalidOrientationModalContent>
                    <PrimaryTextShadow>
                        <Caption align={'center'}>{message}</Caption>
                    </PrimaryTextShadow>
                    <PrimaryFill>
                        <GlassesIcon size={'lg'} style={{ width: 'auto' }} />
                    </PrimaryFill>
                </StyledInvalidOrientationModalContent>
            </StyledInvalidOrientationModalContainer>
        </FadeAnimation>
    </StyledInvalidOrientationNoticel>
);
