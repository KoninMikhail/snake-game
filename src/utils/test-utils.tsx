import React, {PropsWithChildren} from 'react';
import type {RenderOptions} from '@testing-library/react';
import {render} from '@testing-library/react';
import type {PreloadedState} from '@reduxjs/toolkit';
import {configureStore} from '@reduxjs/toolkit';
import {Provider} from 'react-redux';

import type {AppStore, RootState} from '@store/store';
import appSlice from '@store/slices/app.slice';
import soundSlice from '@store/slices/sound.slice';
import maxScoreSlice from '@store/slices/maxScore.slice';
import gameDataSlice from '@store/slices/game.slice';
import {ImagesData} from '@data/imagesData';
import {mockImagesData, mockSoundData} from '@/setupTests';
import SoundsProvider from '@context/SoundsContext';
import ImagesProvider from '@context/ImagesContext';
import {SoundsData} from '@data/soundsData';
import {DefaultTheme, ThemeProvider} from 'styled-components';
import {theme as appTheme} from '@styles/theme';

// As a basic setup, import your same slice reducers

// This type interface extends the default options for render from RTL, as well
// as allows the user to specify other things such as initialState, store.
interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
    preloadedState?: PreloadedState<RootState>;
    store?: AppStore;
    images?: ImagesData;
    sounds?: SoundsData;
    theme?: DefaultTheme;
}

export function renderWithProviders(
    ui: React.ReactElement,
    {
        preloadedState = {},
        // Automatically create a store instance if no store was passed in
        store = configureStore({
            reducer: {
                app: appSlice,
                sounds: soundSlice,
                maxScore: maxScoreSlice,
                game: gameDataSlice,
            },
            preloadedState,
        }),
        images = mockImagesData,
        sounds = mockSoundData,
        theme = appTheme,
        ...renderOptions
    }: ExtendedRenderOptions = {}
) {
    function Wrapper({children}: PropsWithChildren): JSX.Element {
        return (
            <Provider store={store}>
                <ThemeProvider theme={theme}>
                    <SoundsProvider sounds={sounds}>
                        <ImagesProvider images={images}>{children}</ImagesProvider>
                    </SoundsProvider>
                </ThemeProvider>
            </Provider>
        );
    }

    // Return an object with the store and all of RTL's query functions
    return {store, ...render(ui, {wrapper: Wrapper, ...renderOptions})};
}
