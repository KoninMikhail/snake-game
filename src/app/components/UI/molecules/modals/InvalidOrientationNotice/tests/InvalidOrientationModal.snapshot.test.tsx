import {InvalidOrientationModal} from '../InvalidOrientationModal';
import {renderWithProviders} from '@utils/test-utils';

/**
 * Test
 */
describe('InvalidOrientationModal', () => {
    it('equal snapshot', () => {
        const { container } = renderWithProviders(
            <InvalidOrientationModal message={'test container content'} />
        );
        expect(container).toMatchSnapshot();
    });
});
