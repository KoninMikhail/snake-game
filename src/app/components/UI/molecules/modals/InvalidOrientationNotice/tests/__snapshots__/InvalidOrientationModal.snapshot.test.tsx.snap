// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvalidOrientationModal equal snapshot 1`] = `
.c0 {
  display: block;
  position: fixed;
  width: 100%;
  height: 100%;
  top: 0;
  right: 0;
  background: black;
  bottom: 0;
  left: 0;
  z-index: 9111;
}

.c2 {
  display: grid;
  text-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 100%;
  padding: 2.5em;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 9002;
}

.c3 {
  display: grid;
  text-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 1.5rem;
}

.c1 {
  width: inherit;
  height: inherit;
  -webkit-animation-duration: 1400ms;
  animation-duration: 1400ms;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  -webkit-animation-name: cJoqxJ;
  animation-name: cJoqxJ;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-delay: 0ms;
  animation-delay: 0ms;
  -webkit-animation-iteration-count: 1;
  animation-iteration-count: 1;
  -webkit-animation-timing-function: fade;
  animation-timing-function: fade;
}

.c4 {
  font-size: 1.3em;
  font-weight: normal;
  text-align: center;
}

@media (min-width:768px) {
  .c4 {
    font-size: 1.2em;
  }
}

@media (min-width:1200px) {
  .c4 {
    font-size: 1.18em;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
      direction="in"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <span
            class="c4"
            style="text-shadow: 2px 2px 0 deeppink;"
          >
            test container content
          </span>
          <div
            fill="currentColor"
            height="1.8em"
            style="color: deeppink;"
            width="auto"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
